### PR TITLE
chore(Jest): SearchResults [PART-3]

### PR DESCRIPTION
**Pull request by samouss**
This PR moves the folder `test/spec/SearchResults` to Jest. You can run the tests locally with:

```
yarn jest test/spec/SearchResults
```

The migration has been done by [`jest-codemods`](https://github.com/skovhus/jest-codemods) (recommended by the [Jest](https://jestjs.io/docs/en/migration-guide#jest-codemods) project) for all the assertions. Then I removed the dependency on `sinon` file by file. I tried to didn't update the code style or anything unless it was too complicated to migrate.

### CODE REVIEW

File: test/spec/SearchResults/getFacetStats.js [hunk 1]
```diff
@@ -37,43 +36,25 @@ var response = {
...
-    response.results[0].facets_stats.age,
-    'should return the same stats data as in the response');
-  t.deepEqual(
-    result.getFacetStats('price'),
-    response.results[0].facets_stats.price,
-    'should return the same stats data as in the response');
-
-  t.end();
+  expect(bind(result.getFacetStats, result, 'city')).toThrowError(Error);
+  expect(result.getFacetStats('country')).toBe(undefined);
```
Comment (francoischalifour) [inline test/spec/SearchResults/getFacetStats.js]: We can use [`toBeUndefined()`](https://jestjs.io/docs/en/expect#tobeundefined).

File: test/spec/SearchResults/getRefinements.js [hunk 2]
```diff
@@ -44,14 +41,12 @@ test('getRefinements(facetName) returns a refinement(facet) when a facet refinem
...
   }];
 
-  t.deepEqual(refinements, expected);
-  t.equal(refinements.length, refinedFacetValues.length);
-  t.ok(hasSameNames(refinements, refinedFacetValues));
-
-  t.end();
+  expect(refinements).toEqual(expected);
+  expect(refinements.length).toBe(refinedFacetValues.length);
+  expect(hasSameNames(refinements, refinedFacetValues)).toBeTruthy();
```
Comment (francoischalifour) [inline test/spec/SearchResults/getRefinements.js]: We can make the assertion stricter.

File: test/spec/SearchResults/getFacetStats.js (hunk 1 above)
Comment (samouss) [reply test/spec/SearchResults/getFacetStats.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

File: test/spec/SearchResults/getRefinements.js (hunk 2 above)
Comment (samouss) [reply test/spec/SearchResults/getRefinements.js]: See https://github.com/algolia/algoliasearch-helper-js/pull/660#discussion_r278628453

**Pull request merged**